### PR TITLE
Refine isFiberSuspenseAndTimedOut

### PR DESCRIPTION
**Pull request by trueadm**
This changes the `isFiberSuspenseAndTimedOut` implementation based on Sebastian's feedback below:

> I think you might need to check for fiber.memoizedState.dehydrated === null here. If dehydrated is non-null it's a not-yet hydrated boundary. I believe child will be null in that case.

### CODE REVIEW

**Comment by sebmarkbage** [comment]
This might not throw but this still seems like it'll yield the wrong results. The assumptions are still wrong about isFiberSuspenseAndTimedOut

**Comment by trueadm** [comment]
@sebmarkbage What are the correct assumptions? I can adjust accordingly.

File: packages/react-reconciler/src/ReactFiberScope.js [hunk 1]
```diff
@@ -31,7 +31,9 @@ function isFiberSuspenseAndTimedOut(fiber: Fiber): boolean {
 }
```
Comment (sebmarkbage) [inline packages/react-reconciler/src/ReactFiberScope.js]: I think you might need to check for `fiber.memoizedState.dehydrated === null` here.

https://github.com/facebook/react/blob/master/packages/react-reconciler/src/ReactFiberSuspenseComponent.js#L37

If dehydrated is non-null it's a not-yet hydrating boundary.

Comment (trueadm) [reply packages/react-reconciler/src/ReactFiberScope.js]: Addressed your feedback. Thank you :)

**Pull request merged**